doc/ExternalDependencies.md

```diff
@@ -0,0 +1,233 @@
+- board crates: crates in the `boards/` directory
+- capsule crates: crates in the `capsules/` directory
+
+Furthermore, this policy assumes the following rules regarding crate
+dependencies internal to Tock:
+
+- a _board crate_ is not a dependency of any other Tock-internal crate
+- a _chip crate_ is only a dependency of _board crates_ or other
+  _chip crates_
+- a _capsule crate_ is only a dependency of other _capsule crates_ or _board
+  crates_
+- an _arch crate_ may only depend on the _kernel crate_ and other
+  _arch crates_
+- the _kernel crate_ does not depend on _arch_, _chip_, _board_, or _capsule
+  crates_
+
+## External Dependency Selection
+
+External dependencies can be added to Tock on a case-by-case basis. Each
+dependency will be reviewed for inclusion, according to the criteria in this
+section. The requirements are intentionally strict.
+
+There are two general methods to for including an external dependency in the
+Tock kernel: capsule-specific or board-specific external dependencies.
+
+### General Guidelines for Dependency Selection
+
+In general, the following guidelines can provide an indication whether an
+external dependency is suitable for inclusion in Tock.
+
+#### Provide Important Functionality
+
+The external crate provides important functionality that could not easily or
+realistically be provided by the Tock developers.
+
+Such functionality includes:
+
+* Cryptography libraries. Writing cryptographically secure code that is both
+  correct and resistant to attacks is challenging. Leveraging validated,
+  high-quality cryptographic libraries instead of Tock-specific cryptographic
+  code increases the security of the Tock kernel.
+
+#### Project Maturity
+
+The external crate being added should be a mature project, with a high quality
+of code. The project must be well regarded in the Rust community.
+
+#### Limited Sub-dependencies
+
+The external crate should have a limited sub-dependency tree. The fewer
+dependencies the crate introduces the more likely it is to be accepted. There is
+no set threshold, instead this is evaluated on a case-by-case basis.
+
+### Board-Specific External Dependencies
+
+As board crates are generally regarded as use-case specific, managed by specific
+chip and board maintainers, and audited by those maintainers, Tock is more
+flexible with including external dependencies in those crates.
+
+Examples of when a board may want to use an external library:
+
+* Wireless protocols.
+  * Wireless implementations are difficult to get the correct timing.
+  * Wireless protocols are also very expensive to certify.
+
+Note, however, that _only_ the board crate itself may include such an external
+dependency in its `Cargo.toml` file.
+
+A possible way to have other crates indirectly use such a dependency is through
+a wrapper-trait. Such traits abstract the external dependency in a way that
+allows other crates to still be built without the dependency included.
```

Comment:
But we said up above that we want to avoid wrapper-traits. This feels like a contradiction.

Reply:
From our discussion, this point is good, and it's up top that should be made a little more flexible. Use wrappers where they make sense, but not required.